src/amber/router/params.cr

```diff
@@ -56,8 +56,8 @@ module Amber::Router
       # If the current request method is not GET or POST it means that it was
       # already overrided
       return unless %(GET POST).includes? request.method
-      method = params[METHOD]? || override_header?
-      request.method = method if method && OVERRIDE_METHODS.includes? method.upcase
+      method = (params[METHOD]? || override_header?).try &.upcase
```

Comment:
Small recommendation. I didn't realize this till now, but this method has two responsibilities if not three, can we extract the parsing of the `request method` into its own function. Something like

```crystal
def override_request_method!
  return unless %(GET POST).includes? request.method
  request.method = override_method if override_method?
end 

private

def override_method?
  OVERRIDE_METHODS.includes? override_method
end

def override_method
  @_override_method ||= (params[METHOD]? || override_header?).try &.upcase
end
```

What do you think?